infogami/infobase/_dbstore/save.py

```diff
@@ -366,7 +366,7 @@ def compile_index(self, index):
         """Compiles doc-index into db-index.
         """
         keys = set(key for type, key, datatype, name in index)
-        for (type, key, datatype, name), values in index.iteritems():
+        for (type, key, datatype, name), values in index.items():
```

Comment:
Used only once.  Python 2 will build a list in memory which will take up more RAM.  Python 3 will be fast.